relays/parachains/src/parachains_loop.rs

```diff
@@ -246,9 +257,29 @@ where
 			&sync_params.parachains,
 		)
 		.await?;
-		tx_tracker = tx_tracker.take().and_then(|tx_tracker| tx_tracker.update(&heads_at_target));
-		if tx_tracker.is_some() {
-			continue
+
+		// check if our transaction has been mined
+		if let Some(tracker) = submitted_heads_tracker.take() {
```

Comment:
Do you mean wait till our transaction is mined, so something like `tracker.wait().await`? But the state can be changed by some other transaction (e.g. submitted by different relayer) and we may still be waiting for our transaction finality/lose, wasting the time and e.g. losing message delivery rewards because of that. The whole this  transaction tracker is more about waiting for transaction invalidation ("losing" the transaction), not about waiting for its finalization.

Reply:
Oh ok, I see